### PR TITLE
Getting Started Guide for gRPC

### PR DESCRIPTION
**Pull request by danwent**
Signed-off-by: Dan Wendlandt <dan@covalent.io>

Here is a first version of a gRPC getting started guide, including some of the files needed to create the associated demo app container images (not required for running the demo).

### CODE REVIEW

File: Documentation/gettingstarted/grpc.rst [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+.. parsed-literal::
+
+    $ wget |SCM_WEB|\/examples/kubernetes/cilium.yaml
+
+Then make the following changes using a text editor.  
+
+First, change the Cilium container image field from ``cilium/cilium:stable`` to 
+``cilium/cilium:envoy``
+
+Second, add the ``--envoy-proxy`` flag to the args passed to the cilium-agent to indicate that
```
Comment (tgraf) [inline Documentation/gettingstarted/grpc.rst:38]: Needs a single sentence that explains what this flag does and that it enables the tech preview integration of using Envoy as HTTP proxy.

File: Documentation/gettingstarted/grpc.rst [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+
+    // Request maintenance on the door
+    rpc RequestMaintenance(DoorMaintRequest) returns (DoorActionReply) {}
+
+    // Set Access Code to Open / Lock the door
+    rpc SetAccessCode(DoorAccessCodeRequest) returns (DoorActionReply) {}
+
+  }
+
+To keep the setup small, we will just launch just two pods to represent this setup:
```
Comment (tgraf) [inline Documentation/gettingstarted/grpc.rst]: we will just launch just two

File: Documentation/gettingstarted/grpc.rst [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+
+::
+
+    $ kubectl exec terminal-87 -- python3 /cloudcity/cc_door_client.py GetName 1
+    Door name is: Spaceport Door #1
+
+    $ kubectl exec terminal-87 -- python3 /cloudcity/cc_door_client.py GetLocation 1
+    Door location is lat = 10.222200393676758 long = 68.87879943847656
+
+Exposing this information to public terminals seems quite useful, as it helps travels new
```
Comment (tgraf) [inline Documentation/gettingstarted/grpc.rst]: travelers

**Review by tgraf — CHANGES_REQUESTED** [review]

File: Documentation/gettingstarted/grpc.rst [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+
+This document serves as an introduction to using Cilium to enforce gRPC-aware
+security policies.  It is a detailed walk-through of getting a single-node
+Cilium environment running on your machine. It is designed to take 15-30
+minutes.
+
+.. include:: gsg_intro.rst
+.. include:: minikube_intro.rst
+
+Step 1: Installing Cilium
```
Comment (aanm) [inline Documentation/gettingstarted/grpc.rst:13]: On the final step you talk about `minikube delete` maybe we should write to start with minikube. `minikube start --network-plugin=cni`! (I've made the mistake of running it without the --network-plugin)

**Review by aanm — CHANGES_REQUESTED** [review]

File: Documentation/gettingstarted/grpc.rst (hunk 4 above)
Comment (danwent) [reply Documentation/gettingstarted/grpc.rst:13]: @aanm that is what is covered in the text included by .. include minikube_intro.rst

**Comment by danwent** [comment]
@tgraf thanks.  addressed comments.  Since we want this live soon, probably best to push soon and fix up other errors as we get more reviews.   Thx.

**Comment by tgraf** [comment]
Backported into release

File: Documentation/gettingstarted/grpc.rst [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+
+The next step is to install Cilium into your Kubernetes cluster.  Cilium installation
+leverages the `Kubernetes Daemon Set <https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/>`_
+abstraction, which will deploy one Cilium pod per
+cluster node.   This Cilium pod will run in the ``kube-system`` namespace along with
+all other system relevant daemons and services.  The Cilium pod will run both the Cilium
+agent and the Cilium CNI plugin.
+
+Since gRPC runs over HTTP2, Cilium must be run with the Envoy proxy to properly forward
+and filter gRPC requests.   This is not yet the default, we will download and edit
```
Comment (techcet) [inline Documentation/gettingstarted/grpc.rst:24]: *Since* this is not yet the default, ...

File: Documentation/gettingstarted/grpc.rst [hunk 6]
```diff
@@ -0,0 +1,297 @@
...
+the standard cilium daemonset YAML.
```
Comment (techcet) [inline Documentation/gettingstarted/grpc.rst:25]: "Cilium DaemonSet" (or use formatting to look like code or italicize).

File: Documentation/gettingstarted/grpc.rst [hunk 7]
```diff
@@ -0,0 +1,297 @@
...
+and filter gRPC requests.   This is not yet the default, we will download and edit
+the standard cilium daemonset YAML. 
+
+Download Cilium, for example, using wget: 
+
+.. parsed-literal::
+
+    $ wget |SCM_WEB|\/examples/kubernetes/cilium.yaml
+
+Then make the following changes using a text editor.
```
Comment (techcet) [inline Documentation/gettingstarted/grpc.rst:33]: "Then make the following changes to the `cilium.yaml` using a text editor."

File: Documentation/gettingstarted/grpc.rst (hunk 1 above)
Comment (techcet) [reply Documentation/gettingstarted/grpc.rst]: Does that also mean we need to say gRPC is in tech preview earlier on?

File: Documentation/gettingstarted/grpc.rst [hunk 8]
```diff
@@ -0,0 +1,297 @@
...
+.. parsed-literal::
+
+    $ wget |SCM_WEB|\/examples/kubernetes/cilium.yaml
+
+Then make the following changes using a text editor.  
+
+First, change the Cilium container image field from ``cilium/cilium:stable`` to 
+``cilium/cilium:envoy``
+
+Second, add the ``--envoy-proxy`` flag to the args passed to the cilium-agent to indicate that
```
Comment (techcet) [inline Documentation/gettingstarted/grpc.rst:38]: Format in code ``cilium-agent`` or text "Cilium Agent", to be consistent with other parts of our Docs.

File: Documentation/gettingstarted/grpc.rst [hunk 9]
```diff
@@ -0,0 +1,297 @@
...
+    $ kubectl create -f cilium.yaml
+    clusterrole "cilium" created
+    serviceaccount "cilium" created
+    clusterrolebinding "cilium" created
+    configmap "cilium-config" created
+    secret "cilium-etcd-secrets" created
+    daemonset "cilium" created
+
+Kubernetes is now deploying Cilium with its RBAC, ConfigMap and Daemon Set as a
+pod on minkube. This operation is performed in the background.
```
Comment (jrajahalme) [inline Documentation/gettingstarted/grpc.rst:69]: "minkube" -> "minikube"

**Pull request merged**